fairlearn/metrics/_metrics_engine.py

```diff
@@ -3,104 +3,124 @@
 
 import numpy as np
 
-from ._group_metric_result import GroupMetricResult
 from ._input_manipulations import _convert_to_ndarray_and_squeeze
+from sklearn.utils import Bunch
 
 _MESSAGE_SIZE_MISMATCH = "Array {0} is not the same size as {1}"
 
+# Parameters to metrics that should be split according to sensitive features
+_SPLITTABLE_METRIC_PARAMS = {"sample_weight"}
 
-def metric_by_group(metric_function,
-                    y_true, y_pred, group_membership,
-                    sample_weight=None,
-                    **kwargs):
+
+def group_summary(metric_function, y_true, y_pred,
+                  sensitive_features, **metric_params):
     r"""Apply a metric to each subgroup of a set of data.
 
-    :param metric_function: Function ``(y_true, y_pred, sample_weight=None, \*\*kwargs)``
+    :param metric_function: Function ``(y_true, y_pred, \*\*metric_params)``
 
     :param y_true: Array of ground-truth values
 
     :param y_pred: Array of predicted values
 
-    :param group_membership: Array Indicating the group to which each input value belongs
-
-    :param sample_weight: Optional weights to apply to each input value
+    :param sensitive_features: Array indicating the group to which each input value belongs
```

Comment:
It's got to be a single column? If so, shouldn't `sensitive_features` be singular and not plural?

Reply:
i think that our current convention is to call it `sensitive_features` everywhere... for the sake of future compatibility.

Reply:
Perhaps `sensitive_features_column` which is what we settled on for the AzureML integration?

Reply:
I'd rather follow the same convention as we do in `ExponentiatedGradient` and `GridSearch`, where we just use `sensitive_features`.